File: backend/app/schemas/audit.py
"""Response shapes for model-call logs and the admin audit trail."""

from datetime import datetime
from typing import Any

from pydantic import BaseModel


class LlmCallLogOut(BaseModel):
    id: str
    role: str
    provider_name: str
    model: str
    status: str
    latency_ms: int
    error: str | None = None
    created_at: datetime


class AuditLogOut(BaseModel):
    id: str
    actor: str
    action: str
    target: str
    detail: dict[str, Any] | None = None
    created_at: datetime

File: frontend/src/api/audit.ts
import { request } from "./http";

export type CallLog = {
  id: string;
  role: string;
  provider_name: string;
  model: string;
  status: string;
  latency_ms: number;
  error?: string | null;
  created_at: string;
};

export type AuditLog = {
  id: string;
  actor: string;
  action: string;
  target: string;
  detail?: Record<string, unknown> | null;
  created_at: string;
};

export const auditApi = {
  callLogs: () => request<CallLog[]>("/api/admin/logs/calls"),
  auditLogs: () => request<AuditLog[]>("/api/admin/logs/audit"),
  exportLogs: () => fetch("/api/admin/logs/export-errors").then(async (response) => {
    if (!response.ok) throw new Error("导出错误日志失败");
    return response.blob();
  }),
};
